modules/windows-web-app/r-appservice.tf

```diff
@@ -122,6 +122,134 @@ resource "azurerm_windows_web_app" "app_service_windows" {
     }
   }
 
+  dynamic "auth_settings_v2" {
+    for_each = lookup(var.auth_settings_v2, "auth_enabled", false) ? [var.auth_settings_v2] : []
+    content {
+      auth_enabled                            = lookup(auth_settings_v2.value, "auth_enabled", false)
+      runtime_version                         = lookup(auth_settings_v2.value, "runtime_version", "~1")
+      config_file_path                        = lookup(auth_settings_v2.value, "config_file_path", null)
+      require_authentication                  = lookup(auth_settings_v2.value, "require_authentication", null)
+      unauthenticated_action                  = lookup(auth_settings_v2.value, "unauthenticated_action", "RedirectToLoginPage")
+      default_provider                        = lookup(auth_settings_v2.value, "default_provider", "azureactivedirectory")
+      excluded_paths                          = lookup(auth_settings_v2.value, "excluded_paths", null)
+      require_https                           = lookup(auth_settings_v2.value, "require_https", true)
+      http_route_api_prefix                   = lookup(auth_settings_v2.value, "http_route_api_prefix", "/.auth")
+      forward_proxy_convention                = lookup(auth_settings_v2.value, "forward_proxy_convention", "ForwardProxyConventionNoProxy")
+      forward_proxy_custom_host_header_name   = lookup(auth_settings_v2.value, "forward_proxy_custom_host_header_name", null)
+      forward_proxy_custom_scheme_header_name = lookup(auth_settings_v2.value, "forward_proxy_custom_scheme_header_name", null)
+
+      dynamic "apple_v2" {
+        for_each = try(var.auth_settings_v2.apple_v2, null) == null ? [] : [var.auth_settings_v2.apple_v2]
+        content {
+          client_id                  = lookup(apple_v2.value, "client_id", null)
+          client_secret_setting_name = lookup(apple_v2.value, "client_secret_setting_name", null)
+        }
+      }
+
+      dynamic "active_directory_v2" {
+        for_each = try(var.auth_settings_v2.active_directory_v2, null) == null ? [] : [var.auth_settings_v2.active_directory_v2]
```

Comment:
```suggestion
        for_each = var.auth_settings_v2.active_directory_v2[*]
```

Reply:
@jmapro The `try` must be kept

Reply:
Oh you're right ! It can only work if the last value is null and not the map :(